ci/generate-workflows.py

```diff
@@ -90,9 +97,9 @@ def __init__(self, runs_on, container=None, steps=None, env=None):
         self.steps = steps
         self.env = env
 
-    def render(self, f):
+    def render(self, f, depth=3):
```

Comment:
The default value here should be zero to match the definition of `Step.render`; overriding with 3 feels a bit magical to me.

```suggestion
    def render(self, f, depth=0):
```

Reply:
Agree! Probably missed this before my final review. I changed it to 0.